optaplanner-docs/src/main/asciidoc/BenchmarkingAndTweaking/BenchmarkingAndTweaking-chapter.adoc

```diff
@@ -95,7 +95,8 @@ A benchmark configuration XML file looks like this:
 [source,xml,options="nowrap"]
 ----
 <?xml version="1.0" encoding="UTF-8"?>
-<plannerBenchmark>
+<plannerBenchmark xmlns="https://www.optaplanner.org/xsd/benchmark" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="https://www.optaplanner.org/xsd/benchmark https://www.optaplanner.org/xsd/benchmark/benchmark.xsd">
```

Comment:
This is kind of insane. I understand that we want to nudge people towards XSD, and that we want them to copy-paste this, but had I seen this for the first time, I'd run scared.

Solutions? I don't have any. :-(

Reply:
One option I see is taking a shortcut:
- we change any reference to <plannerBenchmark/> and <solver/> to `<plannerBenchmark xmlns="https://www.optaplanner.org/xsd/benchmark">` or `<solver xmlns="https://www.optaplanner.org/xsd/solver">` without the XSD location
- we show the full root element declaration in the Configuration chapter.

As a result, if someone copies such an incomplete declaration into a project, the IDE won't be able to download the XSD from the website. In such a situation IntelliJ IDEA looks for the XSD on the classpath.

Reply:
That could work.

Reply:
@ge0ffrey what is your opinion on this? Which way would you prefer?

Reply:
We need the full declaration as in this PR, I believe. People will just copy paste it, they do it for pom.xml headings too, I am not to worried about it looking scary.

Why not just the "xmlns"? Then the IDE doesn't know where to download the xsd from.
Why not without any xmlns? Then they have no code completion nor xml validation while typing.

Reply:
One could argue that the XSD in xml spec has made that unnecessarily verbose - and I'd agree - but it is what it is.

Reply:
A summary:
- other IDE than IDEA might not be able to handle the incomplete declaration gracefully
- this is how the declaration looks like "by the book"

These points outweigh the concern regarding the rather long and ugly declaration.